components/cmake/find_dep_packages.cmake

```diff
@@ -22,13 +22,16 @@ if (USE_ALBANY OR USE_TRILINOS)
     find_package(Albany REQUIRED)
   endif()
 elseif (USE_KOKKOS)
-  # Kokkos will be built in the sharedlibs if Kokkos_ROOT is
-  # unset.
+  if (NOT DEFINED ENV{Ekat_ROOT})
```

Comment:
This code is not executed if USE_ALBANY=ON or USE_TRILINOS=ON. I think in the above branch we should

1. set Kokkos_ROOT to match Trilinos_ROOT (Albany still uses Trilnos, so we should be able to find trilinos root from albany cmake installation details).
2. call find_package(Ekat REQUIRED)

Reply:
Note: this may be not so relevant now, since I doubt that (currently) we have cases that use Albany AND Ekat at the same time. However, if we ever get landice+eamxx (or maybe landice+omega) in the future, we may get an error due to the fact that ekat's targets are not defined.

I'd rather spend a bit more time now to get the build system robust enough to handle ekat in USE_ALBANY=ON cases.

Reply:
Looking at buildlib.ekat, I see that it is correctly handling the case where USE_ALBANY=ON. Hence, I think we can probably just call `find_package(Ekat REQUIRED)` outside the if blocks, to ensure it is ALWAYS found?

Reply:
Maybe one more caveat: if we have Albany_ROOT _and_ Ekat_ROOT defined, maybe we should run some checks on the ekat targets, to make sure it links against the _same_ kokkos installation that Albany uses?

Reply:
Good point, let me fix this up.